### PR TITLE
implemented and tested customer sign in

### LINKED ISSUE #21: Customer Controller - sign in

### PR DESCRIPTION
**Pull request by Hassan-AbuGareeb**
closed #21

### CODE REVIEW

File: server/server/controllers/customer.js [hunk 1]
```diff
@@ -30,4 +30,19 @@ async function getUsers(req, res) {
   res.json(users);
 }
 
-module.exports = { signup, getUsers };
+async function signin(req, res) {
+  const { email, password } = req.body;
+  try {
+    const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (Shwarzar) [inline server/server/controllers/customer.js:36]: Suggestion to code: Customer's password should have been already hashed during sign up, so during sign in process, bcrypt.compare should be used between the password saved in the database and the password sent in the request body

File: server/server/controllers/customer.js [hunk 2]
```diff
@@ -30,4 +30,19 @@ async function getUsers(req, res) {
...
+    const customer = customerModel.findOne({ email, hashedPassword });
+    if (!customer) throw new Error("wrong username or password");
```
Comment (Shwarzar) [inline server/server/controllers/customer.js:38]: I think we need also to handle incorrect password errors

**Review by Shwarzar — CHANGES_REQUESTED** [review]

File: server/server/controllers/customer.js (hunk 1 above)
Comment (Hassan-AbuGareeb) [reply server/server/controllers/customer.js]: thanks for pointing that out

**Review by Shwarzar — APPROVED** [review]
Approving changes

**Pull request merged**